cdap-gateway/src/main/java/co/cask/cdap/gateway/router/NettyRouter.java

```diff
@@ -289,12 +277,19 @@ private void bootstrapClient(final ChannelUpstreamHandler connectionTracker) {
         new NioClientBossPool(clientBossExecutor, clientBossThreadPoolSize),
         new NioWorkerPool(clientWorkerExecutor, clientWorkerThreadPoolSize)));
 
+
     clientBootstrap.setPipelineFactory(new ChannelPipelineFactory() {
       @Override
       public ChannelPipeline getPipeline() throws Exception {
         ChannelPipeline pipeline = Channels.pipeline();
         pipeline.addLast("tracker", connectionTracker);
         pipeline.addLast("request-encoder", new HttpRequestEncoder());
+        // outbound handler gets dynamically added here (after 'request-encoder')
+        pipeline.addLast("response-decoder", new HttpResponseDecoder());
+        // disable the read-specific and write-specific timeouts; we only utilize IdleState#ALL_IDLE
+        pipeline.addLast("idle-event-generator",
+                         new IdleStateHandler(timer, 0, 0, connectionTimeout));
+        pipeline.addLast("read-timeout", new HttpIdleStateHandler());
```

Comment:
`IdleStateHandler` and `HttpIdleStateHandler` names are too similar and confusing. Maybe call the handler that acts on idle events as `IdleEventProcesor`, or something similar.